src/github_integration/github_manager.py

```diff
@@ -0,0 +1,231 @@
+import logging
+import time
+
+from datetime import datetime
+from typing import Optional
+
+from github import Github
+from github.GitRelease import GitRelease
+from github.RateLimit import RateLimit
+from github.Repository import Repository
+
+from github_integration.model.commit import Commit
+from github_integration.model.issue import Issue
+from github_integration.model.pull_request import PullRequest
+
+
+def singleton(cls):
```

Comment:
why does it need to be singleton?

Reply:
The singleton is used to control GH API calls. There is a need to be very efficient with counting them.
So, one place where to keep all logic and already loaded data.